Update README.md

lazukr opened the pull request:
Updated readme to include JSON format for fireworks display sequence.

lazukr committed:
Update README.md

Updated readme to include JSON format for fireworks display sequence.

jsatgit commented on README.md:
```diff
@@ -58,24 +58,21 @@ This will be the algorithm that will automatically generate a fireworks display
-[Timestamp] TypeOfFirework1 xPosition Angle Speed, TypeOfFirework2 xPosition Angle Speed, etc...
-[Timestamp] ...
-etc...
+{display: [
```
i like this formatting:

```
{
    "key": "value"
}
```

more than
```
{"key": "value"
}
```

lazukr replied:
Done

jsatgit commented on README.md:
```diff
@@ -58,24 +58,21 @@ This will be the algorithm that will automatically generate a fireworks display
-[Timestamp] ...
-etc...
+{display: [
+  {timestamp: 2342342342, shells: [{type: 1, xPosition: 234, angle: 90, speed: 120}]},
```
The keys are strings so put double quotes around them like `"timestamp"`

lazukr replied:
Done

jsatgit commented on README.md:
```diff
@@ -58,24 +58,21 @@ This will be the algorithm that will automatically generate a fireworks display
-etc...
+{display: [
+  {timestamp: 2342342342, shells: [{type: 1, xPosition: 234, angle: 90, speed: 120}]},
+  {timestamp: 12312123, shells: [{type: 1, xPosition: 232, angle: 45, speed: 50}, {type: 2, xPosition: 342, angle: 60, speed: 76}]}
```
Please format large array items as follows:

```
[
   {"type": 1, "xPosition": 232},
   {"type": 2, "xPosition": 242}
]
```

lazukr replied:
Done.

jsatgit commented on README.md:
```diff
@@ -58,24 +58,21 @@ This will be the algorithm that will automatically generate a fireworks display
-[Timestamp] TypeOfFirework1 xPosition Angle Speed, TypeOfFirework2 xPosition Angle Speed, etc...
-[Timestamp] ...
-etc...
+{display: [
```
Why do we have a single top level key called `display`? Are there any other top level keys? If we don't have other top level keys then this object should just be an array to keep it simple.

lazukr replied:
display will be kept as the definition of various shells will be included in the fireworks display sequence.

jsatgit commented on README.md:
```diff
@@ -58,24 +58,21 @@ This will be the algorithm that will automatically generate a fireworks display
-etc...
+{display: [
+  {timestamp: 2342342342, shells: [{type: 1, xPosition: 234, angle: 90, speed: 120}]},
+  {timestamp: 12312123, shells: [{type: 1, xPosition: 232, angle: 45, speed: 50}, {type: 2, xPosition: 342, angle: 60, speed: 76}]}
```
I think the initial speed of a shell is a property of the shell. So shouldn't that be included in the type?

lazukr replied:
Initial speed will be part of the shell property.

jsatgit commented on README.md:
```diff
@@ -58,24 +58,21 @@ This will be the algorithm that will automatically generate a fireworks display
+## Type
 
-This is the specific firework that is to be fired. A database of all existing fireworks will be made, each with a unique identifier. TypeOfFirework will give us details regarding the number of stars used, the colour, type of stars used etc. A TypeOfFirework can be made of TypeOfFirework (the stars are fireworks themselves).
+This is the type of firework that is to be fired. A generic set of fireworks will be available. Custome fireworks can also be made. The fireworks details will also be in JSON format and requires a unique identifier. The type gives us information regarding the number of stars used, the type of stars used, the colour etc. 
```
When you say custom ones can also be made, do you mean we are going to build a firework GUI for people to customize their fireworks? What about the explosion delay? Is that going to be included in the type of firework as well?

lazukr replied:
user defined fireworks will be done using JSON as well. Fireworks themselves are in JSON.

jsatgit replied:
the "actual user" should not have to manipulate the JSON directly. What I'm asking is that, are we planning to build a GUI later on that allows the user to add custom fireworks.

jsatgit commented on README.md:
```diff
@@ -58,24 +58,21 @@ This will be the algorithm that will automatically generate a fireworks display
+## Type
 
-This is the specific firework that is to be fired. A database of all existing fireworks will be made, each with a unique identifier. TypeOfFirework will give us details regarding the number of stars used, the colour, type of stars used etc. A TypeOfFirework can be made of TypeOfFirework (the stars are fireworks themselves).
+This is the type of firework that is to be fired. A generic set of fireworks will be available. Custome fireworks can also be made. The fireworks details will also be in JSON format and requires a unique identifier. The type gives us information regarding the number of stars used, the type of stars used, the colour etc. 
```
We discussed before that frequencies will be mapped to the `xPosition` and amplitude to height of explosion and power of explosion. Here we have no notion of height but we have `explosionDelay`. Can we incorporate that?

lazukr replied:
burstDelay has been included as a property of the shell.

jsatgit commented:
In general please give your branch a more descriptive name than `patch-1`

lazukr committed:
Update README.md

Updated the readme to reflect changes regarding JSON format. Added shell definitions.

jsatgit commented on README.md:
```diff
@@ -58,24 +58,56 @@ This will be the algorithm that will automatically generate a fireworks display
-etc...
+{
+  "shellDefinitions": [
+    {"type":1, "specs": {
```
one key per line please. Also the first key should go on the next line

jsatgit commented on README.md:
```diff
@@ -58,24 +58,56 @@ This will be the algorithm that will automatically generate a fireworks display
+      "timestamp": 2342342342, 
+      "shells": [
+                  {"type": 1, "xPosition": 234, "angle": 90}
+                ]
```
i would prefer to format it like:
```
  "shells": [
    {"type": 1, "xPosition": 234, "angle": 90}
  ]

```

jsatgit commented:
Please wrap json inside of triple back ticks so markdown formats it as code

lazukr committed:
Update README.md

updated JSON formatting.

jsatgit commented:
check this out @lazukr : http://jsonformatter.org/

jsatgit commented on README.md:
```diff
@@ -59,39 +59,44 @@ This will be the algorithm that will automatically generate a fireworks display
+        "size": 10,
+        "burstPower": 20,
+        "burstDelay": 300
+      }
```
there needs to be a comma here

jsatgit replied:
also ur indent is off

lazukr committed:
Update README.md

updated indentations, added a missing comma.

jsatgit commented on README.md:
```diff
@@ -58,24 +58,61 @@ This will be the algorithm that will automatically generate a fireworks display
+        "numOfStars": 10,
+        "typeOfStars": "circular",
+        "colour": "blue",
+        "size": 10,
```
is this the radius of the stars or the radius of the burst?

lazukr replied:
this is size of shell.

jsatgit commented on README.md:
```diff
@@ -58,24 +58,61 @@ This will be the algorithm that will automatically generate a fireworks display
+        "burstDelay": 500
+    }
+  ],
+  "display": [
```
let's use `sequence` instead of `display` ?

lazukr replied:
aye boss.

jsatgit commented on README.md:
```diff
@@ -58,24 +58,61 @@ This will be the algorithm that will automatically generate a fireworks display
+{
+  "shellDefinitions": [
+    {
+      "type":1, 
```
there should be a space between the colon and the 1

lazukr committed:
Update README.md

removed specs key, unnecessary layer. Fixed spacing, renamed size to sizeOfShell for clarity.
